src/Microsoft.AspNetCore.Razor.Language/TagHelperBinderSyntaxTreePass.cs

```diff
@@ -345,7 +345,8 @@ public override void VisitTagHelperPrefixDirectiveSpan(TagHelperPrefixDirectiveC
                 TagHelperDirectiveType directiveType)
             {
                 directiveText = directiveText.Trim();
-                if (directiveText.StartsWith("\"", StringComparison.Ordinal) &&
+                if (directiveText.Length >= 2 &&
```

Comment:
Did you take a look at our other directive parsing? It would be good to do a pass anywhere else we look at directive tokens since the parser doesn't validate them super hard.